src/model/schema.js

```diff
@@ -465,7 +465,7 @@ export class MarkType extends SchemaItem {
 // with node type constructors and another similar object associating
 // mark names with mark type constructors.
 export class SchemaSpec {
-  // :: (?Object<NodeType>, ?Object<MarkType>)
+  // :: (?Object<SubClass<NodeType>>, ?Object<SubClass<MarkType>>)
```

Comment:
I took it from Flow, or, more precisely React: facebook/flow#814. Both TypeScript and Flow use `typeof` to refer to the type of instances of something, so `typeof NodeType` would be `NodeType`, but that doesn't support polymorphism. I actually think this pattern is not too great, since being a sub-class of something doesn't give you a lot of guarantees. I'd rather go with passing builders `Object<()  → NodeType>`, but I don't know if that solves your usecase.